code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+/obj/effect/warped_rune/yellowspace/process()
+	if(cooldown > world.time)
+		return
+
+	var/area/rune_area = get_area(rune_turf)
+	cooldown = world.time + max_cooldown
+	for(var/obj/item/recharged in rune_turf) //recharges items on the rune
+		electrivore(recharged)
+	for(var/obj/machinery/power/apc/apc_recharged in rune_area) //recharges the APC of the room
+		electrivore(apc_recharged)
+
+
+///charge the battery of an item by 20% every time it's called.
+/obj/effect/warped_rune/yellowspace/proc/electrivore(obj/recharged)
+	if(recharged.get_cell()) //check if the item has a cell
+		var/obj/item/stock_parts/cell/battery = recharged.get_cell()
+		if(battery.charge <! battery.maxcharge) //don't charge if the battery is full
+			return
+
+		battery.charge += battery.maxcharge * 0.2
+		battery.update_icon()
+		recharged.update_icon()
+		if(battery.charge > battery.maxcharge)
+			battery.charge = battery.maxcharge
+
+
+/* Dark purple crossbreed, Fill up any beaker like container with 50 unit of plasma dust every 30 seconds  */
+
+
+/obj/item/slimecross/warping/darkpurple
+	colour = "dark purple"
+	runepath = /obj/effect/warped_rune/darkpurplespace
+	effect_desc = "Makes a rune that will periodically create plasma dust,to harvest it simply put a beaker of some kind over the rune."
+
+
+/obj/effect/warped_rune/darkpurplespace
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "plasma_crystal"
+	desc = "The purple ocean would only grow bigger with time."
+	max_cooldown = 300 //creates 50 unit every 30 seconds
+
+
+/obj/effect/warped_rune/darkpurplespace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/darkpurplespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		dust_maker()
+
+
+/obj/effect/warped_rune/darkpurplespace/proc/dust_maker()
+	for(var/obj/item/reagent_containers/glass/beaker in rune_turf)
+		beaker.reagents.add_reagent(/datum/reagent/toxin/plasma,25)
+
+
+/*People who step on the dark blue rune will suddendly get very cold,pretty straight forward.*/
+
+
+/obj/item/slimecross/warping/darkblue
+	colour = "dark blue"
+	runepath = /obj/effect/warped_rune/darkbluespace
+	effect_desc = "Draws a rune creating an unbearable cold above the rune."
+
+
+/obj/effect/warped_rune/darkbluespace
+	desc = "Cold,so cold, why does the world always feel so cold?"
+	icon_state = "cold_rune"
+
+
+/obj/effect/warped_rune/darkbluespace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+	RegisterSignal(rune_turf,COMSIG_ATOM_ENTERED,.proc/cold_tile)
+
+
+/obj/effect/warped_rune/darkbluespace/process() //will keep the person on the tile cold for good measure
+	cold_tile()
+
+
+///it makes people that step on the tile very cold.
+/obj/effect/warped_rune/darkbluespace/proc/cold_tile()
+	for(var/mob/living/carbon/human in rune_turf)
+		human.adjust_bodytemperature(-1000) //Not enough to crit anyone not already weak to cold, might need serious rebalance if cold damage is reworked.
```

Comment:
I never saw the temperature actually go under -1000° even if you continuously stay on the rune because the temperature of the body just adapts incredibly fast for really low temperatures, you'd need to already be really fucking cold to get to absolute zero